dash/dash.py

```diff
@@ -1527,6 +1537,21 @@ def enable_dev_tools(
             _reload.watch_thread.daemon = True
             _reload.watch_thread.start()
 
+        if debug and dev_tools.prune_errors:
+            @self.server.errorhandler(Exception)
+            def _wrap_errors(_):
+                # find the callback invocation, if the error is from a callback
+                # and skip the traceback up to that point
+                # if the error didn't come from inside a callback, we won't
+                # skip anything.
+                tb = get_current_traceback()
+                skip = 0
+                for i, line in enumerate(tb.plaintext.splitlines()):
+                    if "%% callback invoked %%" in line:
+                        skip = int((i + 1) / 2)
```

Comment:
why the skip is about half of code to the callback stub? just curious about the algorithm

Reply:
The plaintext traceback has 2 lines for each stack frame: first line lists the file, line #, and function; second line is the actual line of code. +1 gives the first line of the next pair - the first frame we want to report - then /2 gives the number of frames above it.